### DISCUSSION TITLE
Unable to connect to mcu on Spider 2.2 to Pi4b

### DISCUSSION THREAD

**Discussion by GScully** [FYSETC/FYSETC-SPIDER#92] [Q&A]
Hello, I've followed all the guides to connect my Pi 4b to the Spider 2.2 board via klipper. I've tried the included cable to the pi4b, and a usb c to the pi4b and can not get the systems to work,

G0 X10 F6000
**mcu 'mcu': Unable to connect**
Once the underlying issue is corrected, use the
"FIRMWARE_RESTART" command to reset the firmware, reload the
config, and restart the host software.
Error configuring printer

In the terminal tab i see this logged.

Changing monitoring state from "Opening serial connection" to "Connecting"
Connected to: Serial<id=0xac24c9f0, open=True>(port='/tmp/printer', baudrate=250000, bytesize=8, parity='N', stopbits=1, timeout=10.0, xonxoff=False, rtscts=False, dsrdtr=False), starting monitor
Send: N0 M110 N0*125
Recv: ok
Send: N0 M110 N0*125
Changing monitoring state from "Connecting" to "Operational"

Then if I try any command I get the error Recv: !! mcu 'mcu': Unable to connect 
Yet I can open the 'editor' and view the printer.cfg
That seems odd?
Some more troubleshooting stuff,

pi@voronone:~ $ ls -l /dev/serial/by-id/*
ls: cannot access '/dev/serial/by-id/*': No such file or directory
pi@voronone:~ $ ls -l /dev/serial/by-id/
ls: cannot access '/dev/serial/by-id/': No such file or directory
pi@voronone:~ $ lsusb
Bus 002 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
Bus 001 Device 003: ID 045e:0761 Microsoft Corp. LifeCam VX-2000
Bus 001 Device 002: ID 2109:3431 VIA Labs, Inc. Hub
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
 
Any ideas?

**Comment by ataman78** [comment]
Try gather a write firmware/
